cpp/benchmarks/string/convert_floats_benchmark.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fixture/benchmark_fixture.hpp>
+#include <synchronization/synchronization.hpp>
+
+#include <benchmark/benchmark.h>
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/strings/convert/convert_floats.hpp>
+#include <cudf/types.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+
+#include <numeric>
+#include <random>
+#include <unordered_map>
+
+namespace {
+
+// For each array_size, this function is called twice from both StringToFloatNumber and
+// StringFromFloatNumber classes. Thus, the results are cached for reuse.
+template <class FloatType>
+static const std::vector<FloatType>& get_float_numbers(int64_t array_size)
```

Comment:
Assuming you do the `to_string` conversion on the GPU, I would suggest using `create_random_table` here. The default distribution is Gaussian, which is arguably a more common for real data, and does affect this benchmark. You can still memoize the result.

Reply:
Good idea. I've changed that.

Reply:
I just realized that we cannot use static variable to cache the random table: https://app.slack.com/client/T04SYRAP3/CDTANRCTT/thread/CDTANRCTT-1614629965.099000

So, just generate a random table each time we need it.